Revise introduction section (#42)
greenelab/knowledge-graph-review: content/02.introduction.md

## Introduction

Knowledge graphs are a practical resource for many real-world applications.
They have been used in social network mining to classify nodes [@doi:10.1007/978-1-4419-8462-3_5] and create recommendation systems [@doi:10.1145/3184558.3186904].
These graphs have also been used in natural language processing to interpret simple questions and use relational information to provide answers [@arxiv:1404.4326; @doi:10.1145/3038912.3052675].
In a biomedical setting, these graphs have been used to prioritize genes relevant to disease [@pmid:21121028; @doi:10.1101/651000; @doi:10.1109/CIBCB.2019.8791472; @doi:10.1093/bioinformatics/bty559], perform drug repurposing [@doi:10.7554/eLife.26726] and identify drug-target interactions [@doi:10.1371/journal.pcbi.1002574].

Despite their utility, precisely defining a knowledge graph is a difficult task because there are multiple conflicting definitions [@raw:KG_DEF].

> For this review, we define a knowledge graph as the following: a resource that integrates single or multiple sources of information into the form of a graph where a node repesents entites and an edge represents a relationship between two entites.

Reviewer: typos - entities.

Also, there's nothing in here about direction. Do you want to specify anything about the edges directionality? What definitions did you find elsewhere?

> This graph structure allows for the capacity to make semantic interpretation, continuous incorporation of new information and uncover novel hidden knowledge through computational techniques and algorithms.

Reviewer: I suspect that "semantic interpretation" is going to depend on how you define a knowledge graph above.

> Based on this definition, resources like Hetionet [@doi:10.7554/eLife.26726] would be considered a knowledge graph as Hetionet integrates multiple sources of information into the form of a graph (example shown in Figure {@fig:hetionet_schema}).

Reviewer: Is there any difference between heterogeneous networks and knowledge graphs?

## Introduction

Knowledge graphs are a practical resource for many real-world applications.
They have been used in social network mining to classify nodes [@doi:10.1007/978-1-4419-8462-3_5] and create recommendation systems [@doi:10.1145/3184558.3186904].
These graphs have also been used in natural language processing to interpret simple questions and use relational information to provide answers [@arxiv:1404.4326; @doi:10.1145/3038912.3052675].
In a biomedical setting, these graphs have been used to prioritize genes relevant to disease [@pmid:21121028; @doi:10.1101/651000; @doi:10.1109/CIBCB.2019.8791472; @doi:10.1093/bioinformatics/bty559], perform drug repurposing [@doi:10.7554/eLife.26726] and identify drug-target interactions [@doi:10.1371/journal.pcbi.1002574].

Despite their utility, precisely defining a knowledge graph is a difficult task because there are multiple conflicting definitions [@raw:KG_DEF].

> For this review, we define a knowledge graph as the following: a resource that integrates single or multiple sources of information into the form a graph where nodes repesent entities and edges represent relationships that are shared between two entities.

Reviewer: ```suggestion
For this review, we define a biomedical knowledge graph as the following: a resource that integrates one or more sources of information into a graph where nodes represent biomedical entities and edges represent relationships between two entities.
```
Should "sources of information" be "literature-derived sources of information"? I'm having trouble distinguishing what is more specific here than just being a general graph.

> This definition is consistent with other definitions found in the literature [@doi:10.3233/SW-160218;10.1109/MIC.2019.2928449;@doi:10.1016/j.eswa.2019.112948;@doi:10.1109/MSN.2016.030;10.1109/jproc.2015.2483592;@doi:10.1145/1242572.1242667;@doi:10.1109/TKDE.2017.2754499].

Reviewer: ```suggestion
This definition is consistent with other definitions found in the literature [@doi:10.3233/SW-160218;@doi:10.1109/MIC.2019.2928449;@doi:10.1016/j.eswa.2019.112948;@doi:10.1109/MSN.2016.030;@doi:10.1109/jproc.2015.2483592;@doi:10.1145/1242572.1242667;@doi:10.1109/TKDE.2017.2754499].
```

> Often relationships are considered unidirectional (e.g., a compound treats a disease but a disease cannot treat a compound); however, there are cases where relationships can be considered bidirectional (e.g., a compound resembles another compound or a gene covaries with another gene).

Reviewer: ```suggestion
Often relationships are considered unidirectional (e.g., a compound treats a disease but a disease cannot treat a compound); however, there are cases where relationships can be considered bidirectional (e.g., a compound resembles another compound or a gene interacts with another gene).
```

Reviewer: covaries seems like something that would be expression based, while interacts feels like something that would be more likely derived from literature

> Based on our definition of a knowledge graph, graphs that we consider knowledge graphs may not be suitable for applications such as symbolic reasoning [@doi:10.1007/978-0-585-29599-2_11] but are still applicable for many biomedical applications.

Reviewer: ```suggestion
A subset of graphs that meet our definition of a knowledge graph would be unsuitable for methods such as symbolic reasoning [@doi:10.1007/978-0-585-29599-2_11], but we chose a more permissive definition because we identified numerous uses of these broadly defined graphs in the literature.
```

Reviewer: See what you think of this. I'm not in love with it. 🤷‍♂️

> Overall, knowledge graphs provide a practical means of understanding complex concepts, continuously incorporating new sources of information and uncovering new knowledge through computational techniques and algorithms. 

Reviewer: ```suggestion
```

> Resources like Hetionet [@doi:10.7554/eLife.26726] would be considered a knowledge graph as Hetionet integrates multiple sources of information into the form of a graph (example shown in Figure {@fig:hetionet_schema}).

Reviewer: ```suggestion
For example, Hetionet (Figure {@fig:hetionet_schema}) [@doi:10.7554/eLife.26726] would be considered a biomedical knowledge graph by this definition, and it has been used to identify drug repurposing opportunities [@doi:10.7554/eLife.26726].
```

> Plus, Hetionet has been used to derive novel information concerning unique drug treatments [@doi:10.7554/eLife.26726].

Reviewer: ```suggestion
```

We do not consider databases like DISEASES [@doi:10.1016/j.ymeth.2014.11.020] and DrugBank [@doi:10.1093/nar/gkx1037] to be knowledge graphs.
Although these resources contain essential information, they do not represent their data in the form of a graph.

Knowledge graphs are often constructed from manually curated databases [@doi:10.7554/eLife.26726; 10.1371/journal.pcbi.1002574; @doi:10.1038/s41467-017-00680-8; @doi:10.1186/s12859-016-1336-7;  @doi:10.1016/j.jbi.2011.11.017].
These databases provide previously established information that can be incorporated into a graph.

> For example, a graph using DISEASES [@doi:10.1016/j.ymeth.2014.11.020] as a resource would have genes and diseases as nodes, while edges added between nodes would represent an association between a gene and a disease.

Reviewer: Would this be a biomedical knowledge graph?

Author: According to our definition, this hypothetical graph would be considered a biomedical knowledge graph.

Reviewer: Ok - that was my read too! If you think that's ok, I'm fine with it. I just wanted to make sure I was reading the definition correctly.

Lastly, we conclude on the practicality of knowledge graphs and point out future applications that have yet to be explored.

![
The metagraph (i.e., schema) of the knowledge graph used in the Rephetio project [@doi:10.7554/eLife.26726].

> The authors of this project refer to their resource as a heterogenous network (i.e., hetnet); however, we consider a hetnet to be synonymous to our definition of a knowledge graph.

Reviewer: ```suggestion
The authors of this project refer to their resource as a heterogenous network (i.e., hetnet), and this network meets our definition of a knowledge graph.
```
